### PR TITLE
Make parchment layer test also test the parameter after being reordered

### PR DESCRIPTION
**Pull request by shedaniel**
The reorder method is taken straight from LayeredMappingsDependency: https://github.com/FabricMC/fabric-loom/blob/3ded0964c4282060ca9f50ef9db908d83b30e443/src/main/java/net/fabricmc/loom/configuration/providers/mappings/LayeredMappingsDependency.java#L78-L80

This PR should fail tests.

### CODE REVIEW

File: src/test/groovy/net/fabricmc/loom/test/unit/layeredmappings/ParchmentMappingLayerTest.groovy [hunk 1]
```diff
@@ -38,16 +38,18 @@ class ParchmentMappingLayerTest extends LayeredMappingsSpecification {
...
             def tiny = getTiny(mappings)
+            def reorderedMappings = reorder(mappings)
         then:
             mappings.srcNamespace == "named"
             mappings.dstNamespaces == ["intermediary", "official"]
             mappings.classes.size() == 5747
             mappings.classes[0].srcName.hashCode() == -1112444138 // MojMap name, just check the hash
             mappings.classes[0].getDstName(0) == "net/minecraft/class_2573"
-            mappings.classes[0].methods[0].args[0].srcName == "pStack"
+            mappings.classes[0].methods[0].args[0].srcName == "stack"
```
Comment (modmuss50) [inline src/test/groovy/net/fabricmc/loom/test/unit/layeredmappings/ParchmentMappingLayerTest.groovy]: Please check the hashcode of this. See 2 lines above.

Comment (shedaniel) [reply src/test/groovy/net/fabricmc/loom/test/unit/layeredmappings/ParchmentMappingLayerTest.groovy]: Is there a difference? I will do it anyway tho

Comment (modmuss50) [reply src/test/groovy/net/fabricmc/loom/test/unit/layeredmappings/ParchmentMappingLayerTest.groovy]: Oh my bad it was wrong before anyway. Prob doesn’t matter. Thanks

**Review by modmuss50 — APPROVED** [review]

**Comment by modmuss50** [comment]
I should not have merged this as the tests are now failing.

**Comment by modmuss50** [comment]
Ah my bad, I didnt see the `This PR should fail tests` bit. This needs looking into

**Pull request merged**